README.md

```diff
@@ -5,20 +5,28 @@ The algorithms and solutions in this repo are some of the most frequent question
   
 ---  
 ## What is Javascript?
-Javascript is a web based scripting programming language that allows you to build interactive web pages. Javascript is synchronous in nature, and this behavior is acheived through the use of stacks in browsers V8 engine. It can however be synchronously asynchronous. Asynchronous behavior can be leveraged through the use of Browsers web apis (setTimouts/fetch/Promises) which inturn relies on the queue stack and eventLoop to queue these requests. All in all, Javascripts runtime execution is comprised of: 
-
-1. Stack -- part of browsers v8 engine
-2. Webapis
-3. Queue and eventLoop -- not a part of v8 engine
-
-More on this can be found in the below article.
+Javascript is a web based scripting programming language that allows you to build interactive web pages. Javascript's default synchronous behavior is achieved through the use of stacks in browsers V8 engine.  
+  
+While being synchronous in nature, it can also act asynchronously. Asynchronous behavior can be leveraged through the use of `Promises`, which depend on the Queue, Stack, and EventLoop to order and process these requests.
```

Comment:
```suggestion
  
---  
  
While being synchronous in nature, it can also act asynchronously. Asynchronous behavior can be leveraged through the use of `WebAPIs`, which handle the `Task` separately from the `Call Stack`, allowing the `EventLoop` to continue adding new tasks from the `Task Queue` whenever the `Call Stack` is empty.  
  
The `WebAPI` will send it's result to the `Task Queue`, which the `EventLoop` will eventually pick up and send to the `Call Stack`.  

```